postgres/src/config.rs

```diff
@@ -308,32 +309,51 @@ impl Config {
         self.config.get_channel_binding()
     }
 
-    /// Opens a connection to a PostgreSQL database.
-    pub fn connect<T>(&self, tls: T) -> Result<Client, Error>
+    /// Like `connect`, but `notify_fn` will be invoked with any asynchronous
+    /// notifications or connection errors that arise.
+    ///
+    /// `notify_fn` is invoked from an asynchronous runtime and therefore must
+    /// not block.
+    pub fn connect_with<T, F>(&self, tls: T, mut notify_fn: F) -> Result<Client, Error>
     where
         T: MakeTlsConnect<Socket> + 'static + Send,
         T::TlsConnect: Send,
         T::Stream: Send,
         <T::TlsConnect as TlsConnect<Socket>>::Future: Send,
+        F: FnMut(Result<AsyncMessage, Error>) + Send + 'static,
```

Comment:
I'd rather it take a `AsyncMessage` rather than a `Result`.

Reply:
Sure, done.